### ISSUE TITLE
Record and playback without Mic on ES8388

### ISSUE THREAD

**Issue by gerthalberg** [Ai-Thinker-Open/ESP32-A1S-AudioKit#31]
Does anyone know the configuration for recording and playback from the line-in jack - without having the microphones active also. That is short of desoldering the microphones?

**Comment by gerthalberg** [comment]
Good info, then maybe I can solder a new line in jack to those pads

**Comment by xuhongv** [comment]
did you solve it ?

**Comment by gerthalberg** [comment]
No - Haven't had the time yet.

**Comment by xuhongv** [comment]
we update the readme document.

**Comment by pschatzmann** [comment]
I am working to support this board in my [arduino audio tools project](https://www.pschatzmann.ch/home/2021/12/06/the-ai-thinker-audio-kit-experience-or-nothing-is-right/) and stumbeld over the same problem. After some research I found that this is managed with the  ES8388_DACCONTROL16. If you check in the [original driver](https://github.com/espressif/esp-adf/blob/4634a3220e0a0a7900e7d7a723aeaedde6d1194a/components/audio_hal/driver/es8388/es8388.c
) it is hardcoded:
```
    res |= es_write_reg(ES8388_ADDR, ES8388_DACCONTROL16, 0x00); // 0x00 audio on LIN1&RIN1,  0x09 LIN2&RIN2

```
If you change it to 0x09 you should get what you want...

**Comment by pschatzmann** [comment]
I have the same issue and don't find any update!
If I select AUDIO_HAL_ADC_INPUT_LINE2, I also get both the microphone and line in. With AUDIO_HAL_ADC_INPUT_LINE1 I don't get any input at all. I was double checking Register 10 - but it is set up correctly.

https://www.pschatzmann.ch/home/2021/12/15/the-ai-thinker-audiokit-audio-input-bug/

**Comment by GillesLACAUD** [comment]
Same issue for me :-( . I have route the ADC line2 in the register ADC CONTROL2 and on some board this seem to work fine...but with new board now a have also the MIC in the audio input . It is a ESP32-A1S 2974.
@xuhongv  please could you send a link to this readme document and tell us if we can have only the line input and how to set this.

**Comment by TakesTheBiscuit** [comment]
+1

**Comment by johermohit** [comment]
did anyone solved it yet?

**Comment by GillesLACAUD** [comment]
Yes i have solved this....i have bypass the mic, just sold the 2 pins of the mics.
It is an hardware issue on the board, you can't have only the line in .

**Comment by massa007** [comment]
Would be great if anyone could pls upload an example code for recording and saving to SD! Fighting with this board for a while already...

**Comment by KlausMu** [comment]
Some time ago, this worked for me:
https://randomnerdtutorials.com/esp32-microsd-card-arduino/

Instead of
`if(!SD.begin(5)){`
you need
`#define CS              13`
`#define MOSI            15`
`#define MISO             2`
`#define SCK             14`
`  SPIClass spi = SPIClass(HSPI);`
`  spi.begin(SCK, MISO, MOSI, CS);`
`  if(!SD.begin(CS, spi, 4000000)){`

**Comment by massa007** [comment]
Thanks a lot! I got the SD part working but I’m mainly struggling with utilizing the microphones.. I can play stuff from SD but how do I use the mics?

**Comment by KlausMu** [comment]
Never tried it, but here you can find an example for using microphones:
https://github.com/pschatzmann/arduino-audiokit

**Comment by massa007** [comment]
That’s my missing part!! Awesome :)
Thanks a lot!!

**Comment by pschatzmann** [comment]
The AudioKit is just a HAL for the [Audio Tools Library](https://github.com/pschatzmann/arduino-audio-tools). You will find there much more and examples....

**Comment by Jerzeek** [comment]
There is a great discussion about this problem on @pschatzmann's website: https://www.pschatzmann.ch/home/2021/12/15/the-ai-thinker-audiokit-audio-input-bug/ here someone suggested moving 2 capacitors, I tried this and it worked perfectly!
Thank you Phil!

**Comment by schrab** [comment]
![ESP32-A1S cap mod](https://github.com/user-attachments/assets/d171ff8f-6f8f-4a1b-a8ce-a164eda3c235)
In my case after moving caps there's still signal from line in, so I've also removed two caps marked with blue. Now I have separate stereo line-in and microphones.
Also Line-in jack have swapped L/R channels

**Comment by shemerey** [comment]
> Yes i have solved this....i have bypass the mic, just sold the 2 pins of the mics. It is an hardware issue on the board, you can't have only the line in .

what pins did you solder ?

**Comment by pschatzmann** [comment]
To output the log to a file, just replace the Serial with your file that has been opended for output: 

   AudioToolsLogger.begin(Serial, AudioToolsLogLevel::Debug);

**Comment by GillesLACAUD** [comment]
Hi Anton 


You must sold the two pins of each Mic (In red) 


Best 
Musically 
Gilles/Hansy synth 



https://salmacis.wixsite.com/hansysynth 

----- Mail original -----

De: "Anton Shemerey" ***@***.***> 
À: "Ai-Thinker-Open/ESP32-A1S-AudioKit" ***@***.***> 
Cc: "Gilles Lacaud" ***@***.***>, "Comment" ***@***.***> 
Envoyé: Mercredi 22 Janvier 2025 23:32:39 
Objet: Re: [Ai-Thinker-Open/ESP32-A1S-AudioKit] Record and playback without Mic on ES8388 (Issue #31) 





Yes i have solved this....i have bypass the mic, just sold the 2 pins of the mics. It is an hardware issue on the board, you can't have only the line in . 


what pins did you solder ?
